src/conditionals/programs_with_decision_points/medium/q7/MainTest.java

```diff
@@ -0,0 +1,90 @@
+package conditionals.programs_with_decision_points.medium.q7;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+  // Parsons
+  public Clause[] testSentence() {
+    TestOption.isInputTest = true;
+    TestOption.defaultInput = "12345";
+    return new Clause[] {
+        new StringLiteral("Enter a Javatopia ZIP code:"),
+        new NewLine(),
+        new PlaceHolder(),
+        new PlaceHolder(),
+    };
+  }
+
+  @Override
+  public void runMain() {
+    JavaTopia.main(new String[0]);
+  }
+
+  public static Stream<Arguments> inputProviderValidInput() {
+    return Stream.of(
+        Arguments.of(12345, "This ZIP code belongs to String Avenue"),
+        Arguments.of(14325, "This ZIP code belongs to Character County"),
+        Arguments.of(22111, "This ZIP code belongs to Memory Front"),
+        Arguments.of(24912, "This ZIP code belongs to RAM Ranch"));
+  }
+
+  public static Stream<Arguments> inputProviderOutsideTerritoryInput() {
+    return Stream.of(
+        Arguments.of(23123),
+        Arguments.of(29123),
+        Arguments.of(92123),
+        Arguments.of(18123));
+  }
+
+  public static Stream<Arguments> inputProviderInvalidInput() {
```

Comment:
I would recommend checking the numbers on the boundary of the if statements, that way we will catch an off-by-one error. This applies both ways, so you should check both the numbers in and out on the boundary.